File: agent/commands-generator/src/main/java/org/matilda/commands/di/AnnotationProcessorModule.java
package org.matilda.commands.di;

import dagger.Module;
import dagger.Provides;

import javax.annotation.processing.Filer;
import javax.annotation.processing.ProcessingEnvironment;
import javax.annotation.processing.RoundEnvironment;
import javax.lang.model.element.TypeElement;
import javax.lang.model.util.Elements;
import javax.lang.model.util.Types;
import java.util.Set;

@Module
public class AnnotationProcessorModule {
    private final Set<? extends TypeElement> mAnnotations;
    private final RoundEnvironment mRoundEnvironment;

    private final ProcessingEnvironment mProcessingEnvironment;
    private final boolean mWasRun;

    public AnnotationProcessorModule(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv,
                                     ProcessingEnvironment processingEnvironment, boolean wasRun) {
        mAnnotations = annotations;
        mRoundEnvironment = roundEnv;
        mProcessingEnvironment = processingEnvironment;
        mWasRun = wasRun;
    }

    @Provides
    Set<? extends TypeElement> annotations() {
        return mAnnotations;
    }

    @Provides
    RoundEnvironment roundEnvironment() {
        return mRoundEnvironment;
    }

    @Provides
    ProcessingEnvironment processingEnvironment() {
        return mProcessingEnvironment;
    }

    @Provides
    Filer filer() {
        return mProcessingEnvironment.getFiler();
    }

    @Provides
    Types types() {
        return mProcessingEnvironment.getTypeUtils();
    }

    @Provides
    Elements elements() {
        return mProcessingEnvironment.getElementUtils();
    }

    @Provides
    boolean wasRun() {
        return mWasRun;
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/di/AnnotationProcessorModule.kt
package org.matilda.commands.di

import androidx.room.compiler.processing.ExperimentalProcessingApi
import androidx.room.compiler.processing.XFiler
import androidx.room.compiler.processing.XProcessingEnv
import androidx.room.compiler.processing.XRoundEnv
import dagger.Module
import dagger.Provides

@ExperimentalProcessingApi
@Module
class AnnotationProcessorModule(
    private val mProcessingEnvironment: XProcessingEnv,
    private val mRoundEnvironment: XRoundEnv,
    private val mWasRun: Boolean
) {
    @Provides
    fun roundEnvironment() = mRoundEnvironment

    @Provides
    fun processingEnvironment() = mProcessingEnvironment

    @Provides
    fun filer(): XFiler = mProcessingEnvironment.filer

    @Provides
    fun wasRun() = mWasRun
}
